sql/2024/04/inaccurate-sizes-attribute-impact.sql

```diff
@@ -0,0 +1,106 @@
+# HTTP Archive query to measure impact of inaccurate sizes attributes per <img> for WordPress sites.
+#
+# WPP Research, Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+CREATE TEMPORARY FUNCTION
+  getSrcsetSizesAccuracy(payload STRING)
+  RETURNS ARRAY<STRUCT<sizesAbsoluteError INT64,
+  sizesRelativeError FLOAT64,
+  wDescriptorAbsoluteError INT64,
+  wDescriptorRelativeError FLOAT64,
+  actualSizesEstimatedWastedLoadedPixels INT64,
+  actualSizesEstimatedWastedLoadedBytes FLOAT64,
+  wastedLoadedPercent FLOAT64>>
+  LANGUAGE js AS '''
+try {
+  var $ = JSON.parse(payload);
+  var responsiveImages = JSON.parse($._responsive_images);
+  responsiveImages = responsiveImages['responsive-images'];
+  return responsiveImages.map(({
+    sizesAbsoluteError,
+    sizesRelativeError,
+    wDescriptorAbsoluteError,
+    wDescriptorRelativeError,
+    idealSizesSelectedResourceEstimatedPixels,
+    actualSizesEstimatedWastedLoadedPixels,
+    actualSizesEstimatedWastedLoadedBytes
+  }) => {
+    let wastedLoadedPercent;
+    if ( idealSizesSelectedResourceEstimatedPixels > 0 ) {
+      wastedLoadedPercent = actualSizesEstimatedWastedLoadedPixels / idealSizesSelectedResourceEstimatedPixels;
+    } else {
+      wastedLoadedPercent = null;
+    }
```

Comment:
Better to name this variable something more precise that describes which other data it is based on, at a minimum I think we should include the word "pixels" so that it's clear it has to do with the pixels data and not the bytes data. Maybe `wastedLoadedPixelsPercent`?